### PR TITLE
Added database firewall rule feature.

### PR DESCRIPTION
**Pull request by aznashwan**

### CODE REVIEW

File: management/sql/client.go [hunk 1]
```diff
@@ -85,6 +91,108 @@ func (c SQLDatabaseClient) DeleteServer(name string) error {
...
+	}
+
+	err = xml.Unmarshal(resp, &rules)
+	return rules, err
+}
+
+// UpdateFirewallRule update a firewall rule for an Azure SQL Database server.
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn505707.aspx
+func (c SQLDatabaseClient) UpdateFirewallRule(server, rulename string, params FirewallRuleUpdateParams) error {
```
Comment (ahmetalpbalkan) [inline management/sql/client.go]: `ruleName` would be better

File: management/sql/client.go [hunk 2]
```diff
@@ -85,6 +91,108 @@ func (c SQLDatabaseClient) DeleteServer(name string) error {
...
+	url := fmt.Sprintf(azureCreateFirewallRuleURL, server)
+
+	_, err = c.mgmtClient.SendAzurePostRequest(url, req)
+	return err
+}
+
+// GetFirewallRule gets the details of an Azure SQL Database Server firewall rule.
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn505698.aspx
+func (c SQLDatabaseClient) GetFirewallRule(server, rulename string) (FirewallRuleResponse, error) {
```
Comment (ahmetalpbalkan) [inline management/sql/client.go]: `ruleName` would be better

File: management/sql/client.go [hunk 3]
```diff
@@ -85,6 +91,108 @@ func (c SQLDatabaseClient) DeleteServer(name string) error {
...
+
+	url := fmt.Sprintf(azureUpdateFirewallRuleURL, server, rulename)
+	_, err = c.mgmtClient.SendAzurePutRequest(url, "text/xml", req)
+	return err
+}
+
+// DeleteFirewallRule deletes an Azure SQL Database server firewall rule.
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn505706.aspx
+func (c SQLDatabaseClient) DeleteFirewallRule(server, rulename string) error {
```
Comment (ahmetalpbalkan) [inline management/sql/client.go]: `ruleName` would be better

File: management/sql/entities.go [hunk 4]
```diff
@@ -44,6 +44,43 @@ type ListServersResponse struct {
...
+	StartIPAddress string
+	EndIPAddress   string
+}
+
+// FirewallRuleResponse represents the set of data recieved from
+// an Azure database server firewall rule get response.
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn505698.aspx
+type FirewallRuleResponse struct {
+	XMLName        xml.Name `xml:"http://schemas.microsoft.com/windowsazure ServiceResource"`
```
Comment (ahmetalpbalkan) [inline management/sql/entities.go]: you don't need this XMLName I believe.

**Comment by aznashwan** [comment]
@ahmetalpbalkan: Thanks for the review. Fixed all currently outstanding issues.

**Comment by ahmetalpbalkan** [comment]
LGTM. As I reviewed this on mobile, will let @paulmey take a look and merge.

**Comment by paulmey** [comment]
LGTM

**Comment by paulmey** [comment]
Thanks @aznashwan !

**Pull request merged**